supply/lib/supply/uploader.rb

```diff
@@ -230,14 +355,29 @@ def update_obb(apk_version_code, expansion_file_type, references_version, file_s
     end
 
     def update_track(apk_version_codes)
+      return if apk_version_codes.empty?
+
       UI.message("Updating track '#{Supply.config[:track]}'...")
       check_superseded_tracks(apk_version_codes) if Supply.config[:check_superseded_tracks]
 
-      if Supply.config[:track].eql?("rollout")
-        client.update_track(Supply.config[:track], Supply.config[:rollout] || 0.1, apk_version_codes)
-      else
-        client.update_track(Supply.config[:track], 1.0, apk_version_codes)
+      track_release = AndroidPublisher::TrackRelease.new(
+        name: Supply.config[:version_name],
+        release_notes: [],
+        status: "completed",
+        version_codes: apk_version_codes
+      )
+
+      if Supply.config[:rollout]
+        track_release.status = "inProgress"
+        track_release.user_fraction = Supply.config[:rollout].to_f
       end
+
+      track = AndroidPublisher::Track.new(
```

Comment:
DONT CREATE A NEW TRACK - this will override the current track and its releases???? Fetch an existing track, possible keep releases if `version_codes_to_retain` is used, and then save

Reply:
This was _actually_ okay because Google Play handles some things on their side but reworked anyway to feel less dangerous